pkg/client/fake/client_test.go

```diff
@@ -284,6 +285,118 @@ var _ = Describe("Fake client", func() {
 			Expect(obj.ObjectMeta.ResourceVersion).To(Equal("1"))
 		})
 
+		It("should allow updates with non-set ResourceVersion for a resource that allows unconditional updates", func() {
+			By("Updating a new configmap")
+			newcm := &corev1.ConfigMap{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "v1",
+					Kind:       "ConfigMap",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-cm",
+					Namespace: "ns2",
+				},
+				Data: map[string]string{
+					"test-key": "new-value",
+				},
+			}
+			err := cl.Update(context.Background(), newcm)
+			Expect(err).To(BeNil())
+
+			By("Getting the configmap")
+			namespacedName := types.NamespacedName{
+				Name:      "test-cm",
+				Namespace: "ns2",
+			}
+			obj := &corev1.ConfigMap{}
+			err = cl.Get(context.Background(), namespacedName, obj)
+			Expect(err).To(BeNil())
+			Expect(obj).To(Equal(newcm))
+			Expect(obj.ObjectMeta.ResourceVersion).To(Equal("1"))
+		})
+
+		It("should reject updates with non-set ResourceVersion for a resource that allows unconditional updates", func() {
```

Comment:
nit: a/allows/doesn't allow/

Other than that the PR lgtm

Reply:
Fixed, thanks for catching that!